### ISSUE TITLE
[help] `--enable-exec` breaks `behavior: merge`

### ISSUE THREAD

**Issue by TheRealNoob** [viaduct-ai/kustomize-sops#199]
Hello!  I am having a difficult time getting `kustomize.config.k8s.io/behavior: merge` to work for me.  I suspect I'm probably doing something wrong, maybe somebody could point me in the right direction.

I've created [a small repo](https://github.com/TheRealNoob/random/tree/master/ksops%23199) to make explaining my setup easier.  I haven't encrypted the *.sops.yaml files so that you can see the contents.  If you are to run this, these files have to be encrypted first.  In this file below I show the diff between kustomizing this directory with and without  `--enable-exec` .

https://pastebin.com/JcJzmZUz

You'll notice that with enable-exec kustomize follows a `replace` strategy instead of `merge.  Am I putting the annotation in the incorrect place?

I tested this on my local machine with the below versions, but I also tested in the container image [viaductoss/ksops:v4.2.1](https://hub.docker.com/r/viaductoss/ksops)
```
# kustomize version
v5.0.1+ksops.v4.2.1
```

**Comment by devstein** [comment]
Hi @TheRealNoob thanks for making an issue. I'm not sure if the bot's response is correct (maybe GPT memorized this issue in training or maybe its hallucinating 🤷‍♂️), but I do think it's directionally correct in that the error seems like a kustomize bug.

From a quick search I found one potentially related issue https://github.com/kubernetes-sigs/kustomize/issues/5250 but need to dig deeper figure out what's going on. 

You might want to raise this issue in kustomize and see if they can help. Even if they narrow down the issue to KSOPS that would be useful.

**Comment by TheRealNoob** [comment]
@devstein Thank you.  please forgive my lack of understanding, i'm still new to Kustomize, but isn't `--enable-exec` always required when using ksops?  If so it's not possible to ever use `behavior: merge`?

**Comment by devstein** [comment]
Of course. If I understand your setup correctly, this isn't a bug but incorrect usage.

In the you repo share, it looks like you are adding the merge behavior the a [deployment patch ](https://github.com/TheRealNoob/random/blob/master/ksops%23199/overlays/deployment_volume_patch.sops.yaml#L6), then using [KSOPs to apply the patch](https://github.com/TheRealNoob/random/blob/master/ksops%23199/overlays/ksops-generator.yaml#L11) AND also applying it as [normal patch](https://github.com/TheRealNoob/random/blob/master/ksops%23199/overlays/kustomization.yaml#L9-L10)

When you use the `--enable-exec` flag you are applying to deployment patch twice, which might be why `--enable-exec` "breaks" the merge behavior.

Can you remove the [deployment patch from the KSOPS generator](https://github.com/TheRealNoob/random/blob/master/ksops%23199/overlays/ksops-generator.yaml#L11) and see if that fixes it?

**Comment by TheRealNoob** [comment]
Ah there are two patch files.  One encrypted one not.  I'm doing a normal patch of one and a ksops patch of the other.

When I try commenting out the [proposed line](https://github.com/TheRealNoob/random/blob/master/ksops%23199/overlays/ksops-generator.yaml#L11) it does return valid yaml now - the `image` is defined correctly in the container - but the volume patch doesn't work anymore.
